File: ulbra_2023_2/fundamentos_inteligencia_artificial/aula03/ex4.py
import math

def limite_rapido(x):
    if x > 0:
        return 1
    else:
        return -1
    
def funcao_rampa(x):
    if x < 0:
        return 0
    elif x >= 0 and x <= 1:
        return x
    else:
        return 1
    
def funcao_sigmoide(x):
    if x >= 0:
        return 1 -1 / (1 + math.exp(x))
    else:
        return -1 + 1 / (1 - math.exp(x))
    
def main():
    num_entradas = int(input("Digite o número de entradas: "))
    entradas = [float(input(f"Digite a entrada {i + 1}: ")) for i in range(num_entradas)]
    
    pesos1 = [float(input(f"Digite o peso {i + 1} para o primeiro neurônio: ")) for i in range(num_entradas)]
    pesos2 = [float(input(f"Digite o peso {i + 1} para o segundo neurônio: ")) for i in range(num_entradas)]
    
    opc = int(input("""
MENU:
1 - Limite rapido
2 - Funcao rampa
3 - Funcao Sigmoide
Digite a funcao desejada: """))
    
    if opc == 1:
        saida_neuronio1 = limite_rapido(sum(entrada * peso for entrada, peso in zip(entradas, pesos1)))
        resultado = limite_rapido(saida_neuronio1 * sum(entrada * peso for entrada, peso in zip(entradas, pesos2)))
    elif opc == 2:
        saida_neuronio1 = funcao_rampa(sum(entrada * peso for entrada, peso in zip(entradas, pesos1)))
        resultado = funcao_rampa(saida_neuronio1 * sum(entrada * peso for entrada, peso in zip(entradas, pesos2)))
    elif opc == 3:
        saida_neuronio1 = funcao_sigmoide(sum(entrada * peso for entrada, peso in zip(entradas, pesos1)))
        resultado = funcao_sigmoide(saida_neuronio1 * sum(entrada * peso for entrada, peso in zip(entradas, pesos2)))
    else:
        print("Escolha de função inválida.")
        return
    
    print("Saída final:", resultado)

if __name__ == "__main__":
    main()

File: ulbra_2023_2/fundamentos_inteligencia_artificial/aula03/ex4.js
function limite_rapido(x) {
    if (x > 0) {
        return 1;
    } else {
        return -1;
    }
}

function funcao_rampa(x) {
    if (x < 0) {
        return 0;
    } else if (x >= 0 && x <= 1) {
        return x;
    } else {
        return 1;
    }
}

function funcao_sigmoide(x) {
    if (x >= 0) {
        return 1 - 1 / (1 + Math.exp(x));
    } else {
        return -1 + 1 / (1 - Math.exp(x));
    }
}

function main() {
    const num_entradas = parseInt(prompt("Digite o número de entradas: "));
    
    const entradas = [];
    for (let i = 0; i < num_entradas; i++) {
        entradas.push(parseFloat(prompt(`Digite a entrada ${i + 1}: `)));
    }
    
    const pesos1 = [];
    for (let i = 0; i < num_entradas; i++) {
        pesos1.push(parseFloat(prompt(`Digite o peso ${i + 1} para o primeiro neurônio: `)));
    }
    
    const pesos2 = [];
    for (let i = 0; i < num_entradas; i++) {
        pesos2.push(parseFloat(prompt(`Digite o peso ${i + 1} para o segundo neurônio: `)));
    }
    
    const opc = parseInt(prompt(`
MENU:
1 - Limite rápido
2 - Função rampa
3 - Função Sigmoide
Digite a função desejada: `));
    
    let saida_neuronio1, resultado;
    if (opc === 1) {
        saida_neuronio1 = limite_rapido(entradas.map((entrada, index) => entrada * pesos1[index]).reduce((acc, val) => acc + val, 0));
        resultado = limite_rapido(saida_neuronio1 * entradas.map((entrada, index) => entrada * pesos2[index]).reduce((acc, val) => acc + val, 0));
    } else if (opc === 2) {
        saida_neuronio1 = funcao_rampa(entradas.map((entrada, index) => entrada * pesos1[index]).reduce((acc, val) => acc + val, 0));
        resultado = funcao_rampa(saida_neuronio1 * entradas.map((entrada, index) => entrada * pesos2[index]).reduce((acc, val) => acc + val, 0));
    } else if (opc === 3) {
        saida_neuronio1 = funcao_sigmoide(entradas.map((entrada, index) => entrada * pesos1[index]).reduce((acc, val) => acc + val, 0));
        resultado = funcao_sigmoide(saida_neuronio1 * entradas.map((entrada, index) => entrada * pesos2[index]).reduce((acc, val) => acc + val, 0));
    } else {
        console.log("Escolha de função inválida.");
        return;
    }
    
    console.log("Saída final:", resultado);
}

main();
